core/src/stages/generate_pose.cpp

```diff
@@ -40,13 +40,47 @@
 #include <moveit/planning_scene/planning_scene.h>
 #include <rviz_marker_tools/marker_creation.h>
 
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2/convert.h>
+
+#include <chrono>
+
+namespace {
+// TODO(henningkayser): support user-defined random number engines
```

Comment:
Definitely needs to be provided by the user for controlled determinism.
Please address.